[fix][broker] Resolve the issue of frequent updates in message expiration deletion rate

zjxxzjwang opened the pull request:
### Motivation



### Modifications

At present, the message expiration deletion rate is updated in the getmessageExpiryRate() method, and this rate is updated every time the topic status is obtained. This results in the statistical analysis of message expiration and deletion rates not having a fixed periodicity, making it difficult to represent the true message expiration rate.
### Documentation

Only update the rate in the updateRate() method and only retrieve the value in the getmessageExpiryRate() method

<!-- DO NOT REMOVE THIS SECTION. CHECK THE PROPER BOX ONLY. -->

- [ ] `doc` <!-- Your PR contains doc changes. -->
- [ ] `doc-required` <!-- Your PR changes impact docs and you will update later -->
- [x] `doc-not-needed` <!-- Your PR changes do not impact docs -->
- [ ] `doc-complete` <!-- Docs have been already added -->

### Matching PR in forked repository

PR in forked repository: <!-- ENTER URL HERE -->

<!--
After opening this PR, the build in apache/pulsar will fail and instructions will
be provided for opening a PR in the PR author's forked repository.

apache/pulsar pull requests should be first tested in your own fork since the 
apache/pulsar CI based on GitHub Actions has constrained resources and quota.
GitHub Actions provides separate quota for pull requests that are executed in 
a forked repository.

The tests will be run in the forked repository until all PR review comments have
been handled, the tests pass and the PR is approved by a reviewer.
-->

nodece commented on pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:
```diff
@@ -2457,6 +2457,7 @@ public void updateRates(NamespaceStats nsStats, NamespaceBundleStats bundleStats
                 // Populate subscription specific stats here
                 topicStatsStream.writePair("msgBacklog",
                         subscription.getNumberOfEntriesInBacklog(true));
+                subscription.getExpiryMonitor().updateRates();
```
Unnecessary.

zjxxzjwang replied:
Hello, if not here, how can the message expiration rate be updated

nodece reviewed (approved):
LGTM

jiazhai commented:
This is similar to PR https://github.com/apache/pulsar/pull/19759, but not sure how it was reverted

jiazhai commented:
Get the reverting information: https://github.com/apache/pulsar/pull/24073

jiazhai commented:
@zjxxzjwang  Would you please help add a unit test for this change following PR https://github.com/apache/pulsar/pull/19759. Seems this part of code is easy to be broken, it is better to protect it.

zjxxzjwang commented:
> @zjxxzjwang Would you please help add a unit test for this change following PR #19759. Seems this part of code is easy to be broken, it is better to protect it.

Good suggestion, I have also considered it before. But it seems that there are no good test examples to determine if this has not been broken.

zjxxzjwang commented:
@lhotari Hello, can you review it

nodece commented:
@zjxxzjwang Please rebase the master branch, and then I will merge this PR after CI passes.

HQebupt commented:
/pulsarbot rerun-failure-checks

zjxxzjwang commented:
> @zjxxzjwang Please rebase the master branch, and then I will merge this PR after CI passes.

hello , I have rebased the master branch and added test cases. Can you help me merge it
